lib/backend/firestore/firestorebk.go

```diff
@@ -497,7 +497,7 @@ func (b *Backend) Delete(ctx context.Context, key []byte) error {
 	if !doc.Exists() {
 		return trace.NotFound("key %s does not exist", string(key))
 	}
-	_, err = docRef.Delete(ctx)
+	_, err = docRef.Delete(ctx, firestore.Exists)
```

Comment:
Why do we need the `Get` if deleting with a precondition of `Exists` is a thing? Can't we just figure out the error returned by the `Delete` (presumably a grpc NotFound error?) and convert that to our `trace.NotFound("key %s does not exist", string(key))`?

Reply:
We don't
Removing it

Reply:
Is the returned error converted to something for which `trace.IsNotFound()` is true? We rely on that across the whole stack I think

Reply:
Yes, it returns `trace.NotFound`, so `trace.IsNotFound` should evaluate to true
(I did a ninja edit in the commit, sorry)